src/Microsoft.TemplateEngine.Core/VariableCollection.cs

```diff
@@ -284,7 +284,7 @@ public static VariableCollection VariableCollectionFromParameters(IEngineEnviron
                 {
                     if (param.Priority != TemplateParameterPriority.Optional && param.Priority != TemplateParameterPriority.Suggested)
                     {
-                        while(!environmentSettings.Host.OnParameterError(param, null, "ParameterValueNotSpecified", out string val))
+                        while(environmentSettings.Host.OnParameterError(param, null, "ParameterValueNotSpecified", out string val))
```

Comment:
Actually, looking at this again, this will now fail if OnParameterError is hooked up & returns `true` (in cases where the user does supply a valid value) - there should either be a second condition in the while that validates the parameter or this should just be a call without check to the return value.